### PR TITLE
REF bool order2 -> int max_order in Scattering2D

### PR DESCRIPTION
**Pull request by lostanlen**
a contribution to solving #196 in the case of 2D
more consistency between 1D and 2D
enabling the development of third-order scattering in the future

### CODE REVIEW

File: kymatio/scattering1d/scattering1d.py [hunk 1]
```diff
@@ -178,8 +178,7 @@ class Scattering1D(object):
         Required precision for the periodization. Defaults to `1e-7`.
     max_order : int, optional
         The maximum order of scattering coefficients to compute. Must be one
-        `1` or `2`. This parameter may be modified after object creation.
-        Defaults to `2`.
+        `1` or `2`. Defaults to `2`.
```
Comment (janden) [inline kymatio/scattering1d/scattering1d.py]: This should be in a separate PR (or at least a separate commit).

File: kymatio/scattering2d/scattering2d.py [hunk 2]
```diff
@@ -48,6 +48,9 @@ class Scattering2D(object):
         spatial support (M, N) of the input
     L : int, optional
         number of angles used for the wavelet transform
+    max_order : int, optional
+        The maximum order of scattering coefficients to compute. Must be one
```
Comment (janden) [inline kymatio/scattering2d/scattering2d.py]: "Must be one" -> "Must be one of"

File: kymatio/scattering2d/scattering2d.py [hunk 3]
```diff
@@ -61,6 +64,9 @@ class Scattering2D(object):
...
```
Comment (janden) [inline kymatio/scattering2d/scattering2d.py]: Missing "of" (see above).

File: kymatio/scattering1d/scattering1d.py (hunk 1 above)
Comment (lostanlen) [reply kymatio/scattering1d/scattering1d.py]: ok. I'll revert this and make a separate PR.
I'd like to think of `max_order` as an immutable and then maybe in the future we could have `orders` as mutable, which defaults to `range(1+max_order)` and needs to be a subset thereof. this is why i removed this sentence: i did not want to offer the guarantee that `max_order` will remain mutable

Comment (janden) [reply kymatio/scattering1d/scattering1d.py]: That makes sense. It should be tied into the filter generation, after all.

Not here, but we may want to rename this to `_max_order` so people won't mess with it, but I'm not sure how these things are normally done. There's no encapsulation in Python, right?

Comment (lostanlen) [reply kymatio/scattering1d/scattering1d.py]: that's right. nothing is private in Python, but naming an object attribute with a leading underscore is widespread convention for scaring away the user to change these values after construction. the same can be said of methods
(and for C numba bindings, the convention is *two* leading underscores! :D )

Comment (janden) [reply kymatio/scattering1d/scattering1d.py]: Ok.
